etcdserver/server.go

```diff
@@ -570,12 +576,53 @@ type etcdProgress struct {
 	appliedi  uint64
 }
 
+// readyHandler contains a set of EtcdServer operations to be called by raftNode,
+// and helps decouple state machine logic from Raft algorithms.
+// TODO: add a state machine interface to apply the commit entries and do snapshot/recover
```

Comment:
I don't understand why this is a TODO? I think the channel interface for committing entries is OK.